### DISCUSSION TITLE
Proposal for new menu layout

### DISCUSSION THREAD

**Discussion by joeclifford** [Jyers/Marlin#388] [Ideas]
So, this idea has its beginnings in this comment - https://github.com/Jyers/Marlin/pull/310#issuecomment-810303203

I tested the brightness control in the root of the Control menu and it felt out of place; it currently works better in the Advanced menu. However, I agree with @tititopher68's point that it doesn't really belong in the Advanced menu either.

This got me thinking about all the other menu options and it feels like there is no consistency with the layout of the menus. Features are getting implemented and the menu item is getting placed where the author of that feature chose to put it. An example is the Z offset setting being in the prepare menu when most other settings are in the Control menu.

My proposal it that the menu structure is refactored to be more like the stock marlin menu.

https://marlinfw.org/docs/features/lcd_menu.html#lcd-menu-tree

I see 3 main advantages to following the main Marlin layout:

1. Users already familiar with Marlin will be used to the layout
2. It gives guidance for where new features should go in the menu system
3. If the default text based MarlinUI is ported to the DWIN screen, it will be more consistent https://github.com/MarlinFirmware/Marlin/pull/19371

Something like this:
(I don't use manual mesh / BLTouch, so would welcome input for the Level menu; this is based on manual leveling)

```
|--Print
|--Prepare
|   |--Move
|   |   |--Move X
|   |   |--Move Y
|   |   |--Move Z
|   |   |--Auto Home
|   |   |--Extruder
|   |--Auto Home
|   |--Auto Home X
|   |--Auto Home Y
|   |--Auto Home Z
|   |--Manual Leveling
|   |--Disable Steppers
|   |--Change Filament
|   |--Cooldown
|   |--Preheat
|   |   |--Bed Only Mode
|   |   |--Preheat 1
|   |   |--Preheat 2
|   |   |--Preheat 3
|   |   |--Preheat 4
|--Control
|   |--Temperature
|   |   |--Hotend
|   |   |--Bed
|   |   |--Fan
|   |   |--PID Autotune
|   |   |--Preheat 1 Conf
|   |   |--Preheat 2 Conf
|   |   |--Preheat 3 Conf
|   |   |--Preheat 4 Conf
|   |--Motion
|   |   |--Z Offset
|   |   |--Max Speed
|   |   |--Max Acceleration
|   |   |--Max Jerk
|   |   |--Steps/mm
|   |   |--Flow Rate
|   |--Filament
|   |   |--Load Length
|   |   |--Unload Length
|   |   |--Filament Sensor
|   |   |--Runout Distance
|   |--LCD Brightness
|   |--BLTouch   (If/When implemented as a menu item https://github.com/Jyers/Marlin/discussions/367)
|   |--Advanced
|   |   |--Min Extrusion T
|   |   |--Power-loss Recovery
|   |   |--Progress as __h__m
|   |--Store Settings
|   |--Restore Settings
|   |--Reset to Defaults
|   |--Info
|--Info

|--Tune
    |--Print Speed
    |--Hotend
    |--Bed
    |--Fan
    |--Flow Rate
    |--Z Offset
    |--Change Filament
    |--LCD Brightness
```

**Comment by Jyers** [comment]
I like a lot of this. I don't know if I will implement this exact layout, but I think you'll definitely see some adjustments based on it in the future. One of the major things with the layout, in case you're interested, is also consistency with the stock creality firmware. For instance I guarantee if I put the Z-Offset menu all the way in the motion menu I'm going to receive about 20 discussions/reddit comments asking if there is a z offset tool and where it is.
